### DISCUSSION TITLE
Steps for the backend

### DISCUSSION THREAD

**Discussion by lbineau** [aem-vite/vite-aem-plugin#7] [Q&A] [answered]
On the first page of the documentation the plugin looks magic and plug and play.
But after continuing to read it seems there is a quite a lot to modify on the backend requiring a certain amount of knowledge.
I was wondering if all steps of the backend necessary, especially this sentence is confusing me (https://www.aemvite.dev/guide/front-end/vite-plugin/#solving-challenges):
> The Java side of AEM Vite works quite well in small doses but it has some flaws which makes maintaining the replacement logic complex and time-consuming. 

Does someone have a working example with a skeleton of AEM and the modifications you did to make it work with the Vite AEM plugin ?

**Comment by cshawaus** [comment]
Hi @lbineau,

I should probably update that statement as it was included before I completely removed the older replacement code from the plugins. With that in mind, the backend changes are quite straightforward and are contained to just a few pages.

- https://www.aemvite.dev/guide/installation/
- https://www.aemvite.dev/guide/backend/clientlibs/
- https://www.aemvite.dev/guide/backend/permissions/

While I would love for this to be more plug and play it unfortunately cannot be due to the various unknown combinations to which projects can be created. The guides have been simplified to explain just enough to get things working and anything more beyond that is your choice as to how you would like to accomplish the integration.

I did plan on finishing the archetype project a while back but life has gotten in the way and that has had to sit on the back burner for a little while now. If I happen to spot a free moment I see about creating a demo project to highlight the integration but I cannot make any promises.

**Reply by cshawaus** [reply, marked as answer]
The docs have been updated to better reflect the intent behind the Vite plugin.

https://www.aemvite.dev/guide/front-end/vite-plugin/
